docs/zarr-v3.md

```diff
@@ -0,0 +1,96 @@
+# zarr-v3 Guide/Roadmap
+
+`anndata` now uses the much improved {mod}`zarr` v3 package and also allows writing of datasets in the [v3 format], with the exception of structured arrays.
+Users should notice a significant performance improvement, especially for cloud data, but also likely for local data as well.
+Here is a quick guide on some of our learnings so far:
+
+## Remote data
+
+We now provide the {func}`anndata.experimental.read_lazy` feature for reading as much of the {class}`~anndata.AnnData` object as lazily as possible, using `dask` and {mod}`xarray`.
+Please note that this feature is experimental and subject to change.
+To enable this functionality in a performant and feature-complete way for remote data sources, we use [consolidated metadata] on the `zarr` store (written by default).
+Please note that this introduces consistency issues - if you update the structure of the underlying `zarr` store i.e., remove a column from `obs`, the consolidated metadata will no longer be valid.
+Further, note that without consolidated metadata, we cannot guarantee your stored `AnnData` object will be fully readable.
+And even if it is fully readable, it will almost certainly be much slower to read.
+
+There are two ways of opening remote [`zarr` stores] from the `zarr-python` package, `fsspec` and `obstore`, and both can be used with `read_lazy`.
+[`obstore` claims] to be more performant out-of-the-box, but notes that this claim has not been benchmarked with the `uvloop` event loop, which itself claims to be 2X more performant than the default event loop for `python`.
+
+## Local data
+
+Local data generally poses a different set of challenges.
+First, write speeds can be somewhat slow and second, the creation of many small files on a file system can slow down a filesystem.
+For the "many small files" problem, `zarr` has introduced [sharding] in the v3 file format.
+Sharding requires knowledge of the array element you are writing, though, and therefore you will need to use {func}`anndata.experimental.write_dispatched` to use sharding.
+Here is a short example, although you should tune the sizes to your own use-case and also use the compression that makes the most sense for you:
+
+```python
+import anndata as ad
+from collections.abc import Mapping
+from typing import Any
+import zarr
+
+ad.settings.zarr_write_format = 3 # Absolutely crucial! Sharding is only for the v3 file format!
+
+def write_sharded(group: zarr.Group, adata: ad.AnnData):
+    def callback(func: ad.experimental.Write, g: zarr.Group, k: str, elem: ad.typing.RWAble, dataset_kwargs: Mapping[str, Any], iospec: ad.experimental.IOSpec):
```

Comment:
this is very wide and pokes out of the code box in the rendered doc.

Please reformat this (e.g. by pasting into https://play.ruff.rs/) so it looks good on the [rendered page](https://icb-anndata--1948.com.readthedocs.build/en/1948/zarr-v3.html). 

For later: maybe we should try out https://github.com/Freed-Wu/mdformat-ruff? I was against including it in the scverse template since mdformat doesn’t really support MyST, but we should be alert enough to spot when it breaks things :hand_with_index_finger_and_thumb_crossed: